test/live_test.sh

```diff
@@ -412,13 +440,34 @@ del_site_attr() {
 	run_site $site geostore delete $1
 }
 break_external_prog() {
-	run_site $1 crm configure "location $PREFNAME `get_rsc` rule -inf: defined \#uname"
+	local cmd
+	case "$cli_driver" in
+	native) cmd="cibadmin --create --scope constraints --xml-text=\""
+	        cmd="$cmd<rsc_location id='$PREFNAME' rsc='`get_rsc`'><rule id='rule_$PREFNAME' score='-INFINITY'>"
+		cmd="$cmd<expression id='expr_$PREFNAME' attribute='#uname' operation='define'/>"
+		cmd="$cmd</rule></rsc_location>\"";;
+	crm)    cmd="crm configure \"location $PREFNAME `get_rsc` rule -inf: defined \#uname\"";;
+	pcs)    cmd="pcs constraint location `get_rsc` rule id=$PREFNAME score=-INFINITY defined \#uname";;
+	esac
+	runcmd $1 $cmd
 }
 show_pref() {
-	run_site $1 crm configure show $PREFNAME > /dev/null
+	local cmd
+	case "$cli_driver" in
+	native) cmd="cibadmin --query --xpath \"//rsc_location[@id = '$PREFNAME']\"";;
+	crm)    cmd="crm configure show $PREFNAME > /dev/null";;
+	pcs)    cmd="pcs constraint show --full | grep -q $PREFNAME";;
+	esac
+	runcmd $1 $cmd
 }
 repair_external_prog() {
-	run_site $1 crm configure delete $PREFNAME
+	local cmd
+	case "$cli_driver" in
+	native) cmd="cibadmin --delete --xpath \"//rsc_location[@id = '$PREFNAME']\"";;
+	crm)    cmd="crm configure delete $PREFNAME";;
+	pcs)    cmd="pcs constraint remove $PREFNAME";;
+	esac
+	runcmd $1 $cmd
```

Comment:
It definitely needs more consideration, e.g., the `cli_driver` should be first tested across all the machines under test.